src/main/java/org/opensearch/flowframework/workflow/Workflow.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.workflow;
+
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.flowframework.template.Template;
+import org.opensearch.flowframework.template.WorkflowEdge;
+import org.opensearch.flowframework.template.WorkflowNode;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+
+/**
+ * This represents an object in the workflows section of a {@link Template}.
+ */
+public class Workflow implements ToXContentObject {
+
+    /** The template field name for workflow user params */
+    public static final String USER_PARAMS_FIELD = "user_params";
+    /** The template field name for workflow nodes */
+    public static final String NODES_FIELD = "nodes";
+    /** The template field name for workflow steps, an alternative name for nodes */
+    public static final String STEPS_FIELD = "steps";
```

Comment:
Can `nodes` and `steps` be used interchangeably? If so, why not just use `nodes`.

Reply:
Why not allow either?  I'm ok, just was looking at the sample template that had both.

Reply:
Sample template has been updated, as per @ohltyler let's stick to `nodes` so we are not using language that infers there cannot be graphs.

Reply:
Let's just have `nodes`?